src/layers/src/base-layer.ts

```diff
@@ -267,10 +279,24 @@ class Layer {
   }
 
   get requiredLayerColumns(): string[] {
+    const {supportedColumnModes} = this;
```

Comment:
Nit: This is obviously totally out of scope for this PR, but seems there are some opportunities for improvement here.
- Add a comment over each method?
- Is there a reason we define accessors? I don't see why this is not just `getRequiredLayerColumns()`